src/ProjectMWrapper.cpp

```diff
@@ -59,6 +61,42 @@ void ProjectMWrapper::initialize(Poco::Util::Application& app)
 
         _projectM = projectm_create_settings(&settings, PROJECTM_FLAG_NONE);
 
+        // remove any presets in blocklist.txt
+        std::ifstream infile("blocklist.txt");
+        std::string line;
+        for( std::string line; getline( infile, line ); )
+        {
+            unsigned int size = projectm_get_playlist_size(_projectM);
+            unsigned int index = projectm_get_preset_index(_projectM, line.c_str());
+            if (index < size) {
+                projectm_remove_preset(_projectM, index);
+                SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "INIT preset block '%s'\n", line.c_str());
+            }
+        }
+
+        // remove any presets that don't contain filter
+        if (presetFilter != "") {
+            SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "INIT preset filter '%s'\n", presetFilter.c_str());
+
+            unsigned int size = projectm_get_playlist_size(_projectM);
+
+            int offset = 0;
+            for (int i = 0; i < size; i++) {
```

Comment:
is it dangerous to modify a list you're iterating over? will size be too large after removing some items?

Reply:
fixed by iterating backwards